### ISSUE TITLE
Pressing "Access Server" as admin leads to ones personal account

### ISSUE THREAD

**Issue by mrocklin** [jupyterhub/zero-to-jupyterhub-k8s#525]
When diagnosing a user's issue I log into the hub/admin page and press the "Access Server" button next to their account name.  Instead of being taken to their account I am taken to my own.

## Contents of `config.yaml`

    --version=v0.6.0-9701a90

```
proxy:
  secretToken: SECRET

singleuser:
  image:
    name: daskdev/pangeo-notebook
    tag: 2018-02-14
  cmd: ['start-singleuser.sh']
  extraEnv:
    EXTRA_PIP_PACKAGES: >-
    GCSFUSE_BUCKET: pangeo-data
  storage:
    extraVolumes:
      - name: fuse
        hostPath:
          path: /dev/fuse
    extraVolumeMounts:
      - name: fuse
        mountPath: /dev/fuse
  cpu:
    limit: 2
    guarantee: 1
  memory:
    limit: 4G
    guarantee: 2G


# proxy:
#   https:
#     hosts:
#       - pangeo.pydata.org
#     letsencrypt:
#       contactEmail: mrocklin@gmail.com

auth:
  type: github
  github:
    clientId: "2cb5e09d5733ff2e6ae3"
    clientSecret: "SECRET"
    callbackUrl: "http://pangeo.pydata.org/hub/oauth_callback"
  admin:
    access: true
    users:
      - mrocklin
      - jhamman
      - rabernat
      - yuvipanda
      - choldgraf

rbac:
  enabled: false

hub:
  extraConfig: |
    c.KubeSpawner.singleuser_service_account = 'default'
    c.Spawner.default_url = '/lab'
    from kubernetes import client
    def modify_pod_hook(spawner, pod):
        pod.spec.containers[0].security_context = client.V1SecurityContext(
            privileged=True,
            capabilities=client.V1Capabilities(
                add=['SYS_ADMIN']
            )
        )
        return pod
    c.KubeSpawner.modify_pod_hook = modify_pod_hook

cull:
  enabled: true
  users: false
  timeout: 3600
  every: 600
```

**Comment by rabernat** [comment]
I experienced this too. But then I tried it again a few times and it worked.

**Comment by consideRatio** [comment]
+1

**Comment by willingc** [comment]
cc/ @minrk

**Comment by minrk** [comment]
Was the user's server running at the time you requested access? My guess is you are running into [this sort of event](https://github.com/jupyterhub/jupyterhub/issues/1629#issuecomment-365259056) which I'm currently working on in https://github.com/jupyterhub/jupyterhub/pull/1660.

**Comment by mrocklin** [comment]
When running into this I did typically click "start server" and then "access server" directly afterwards.  I would not be surprised to learn that the pod was still spinning up when I pressed "access server".

**Comment by consideRatio** [comment]
@mrocklin I think so too, this happens when the server is about to start, but has not actually done it fully.

I guess, that what happens is that the admin visits another user page but that user page is in the "server is starting up..." state.

**Comment by minrk** [comment]
Yeah, access server definitely results in the wrong behavior if the intended server isn't finished starting up. This will be fixed by jupyterhub/jupyterhub#1660.

**Comment by consideRatio** [comment]
Wieee nice fixing!
